maintenance/wikia/email.php

```diff
@@ -0,0 +1,200 @@
+		);
+		$this->addOption( 'list', 'List email types',
+			Maintenance::PARAM_OPTIONAL,
+			Maintenance::PARAM_IS_FLAG
+		);
+	}
+
+	public function execute() {
+		try {
+			$this->initOptions();
+
+			if ( $this->list ) {
+				$this->showValidEmailTypes();
+				exit( 0 );
+			}
+
+			// The controller class should end with 'Controller'
+			$controllerClass = 'Email\\Controller\\' . $this->type . 'Controller';
+			$this->assertControllerExists( $controllerClass );
+
+			$params = $this->argStringToArray( $this->argString );
+			$resp = F::app()->sendRequest( $controllerClass, 'handle', $params, true );
+
+			$this->assertSuccessfulEmail( $resp );
+
+			echo "Email sent as '$this->user' successfully!\n";
+			exit( 0 );
+		} catch ( EmailCLIException $e ) {
+			echo "An error occurred:\n";
+			echo $e->getMessage() . "\n";
+			exit( 1 );
+		}
+	}
+
+	protected function initOptions() {
+		$this->test = $this->hasOption( 'test' );
+		$this->verbose = $this->hasOption( 'verbose' );
+		$this->user = $this->getOption( 'user', '' );
+		$this->type = $this->getOption( 'type', '' );
+		$this->argString = $this->getOption( 'args', '' );
+		$this->list = $this->getOption( 'list', false );
+
+		$this->assertRequiredOptionsExist();
+
+		// The list functionality doesn't need an email address
+		if ( !$this->list ) {
+			$this->setupUser();
+		}
+	}
+
+	protected function assertRequiredOptionsExist() {
+		if ( $this->list ) {
+			return;
+		}
+
+		if ( !empty( $this->user ) && !empty( $this->type ) ) {
+			return;
+		}
+
+		throw new OptionException( 'Either --list or both --to and --type options are required' );
+	}
+
+	protected function setupUser() {
+		// We either have a user ID or a user name
+		if ( filter_var( $this->user, FILTER_VALIDATE_INT ) ) {
+			$userObject = User::newFromId( $this->user );
+		}
+
+		// Try this as a user name next if we still didn't find what we want.
+		if ( empty( $userObject ) ) {
+			$userObject = User::newFromName( $this->user );
+		}
+
+		if ( empty( $userObject ) ) {
+			throw new OptionException( 'Could not find matching user for name/ID: ' . $this->user );
+		}
+
+		F::app()->wg->User = $userObject;
+	}
+
+	protected function argStringToArray( $argString ) {
+		$paramPairs = explode( ',', $argString );
+
+		$param = [];
+		foreach ( $paramPairs as $pair ) {
+			if ( empty( $pair ) ) {
+				continue;
+			}
+
+			list( $key, $value ) = explode( '=', $pair );
+			if ( empty( $value ) ) {
+				echo "WARNING: argument '$key' had no value\n";
+				continue;
+			}
+			if ( empty( $key ) ) {
+				echo "WARNING: missing key for value '$value'\n";
+				continue;
+			}
+			$param[$key] = $value;
+		}
+
+		return $param;
+	}
+
+	protected function assertControllerExists( $controllerClass ) {
+		if ( !class_exists( $controllerClass ) ) {
+			throw new OptionException( "Can't find $controllerClass to handle this email type\n" );
+		}
+	}
+
+	protected function assertSuccessfulEmail( WikiaResponse $resp = null ) {
```

Comment:
Is there a reason this needs a default argument? This is only called in one place and always passes in an argument.